Utilities/CodeGeneration/EdFi.Ods.CodeGen/Generators/Resources/ResourceProfileData.cs

```diff
@@ -292,7 +292,7 @@ public ResourceChildItem GetContainedResource(ResourceClassBase resource)
 
             return SuppliedResource != null
                 ? SuppliedResource.AllContainedItemTypes.FirstOrDefault(
-                    x => ModelComparers.Entity.Equals(x.Entity, resource.Entity))
+                    x => ModelComparers.Resource.Equals(x, resource) && ModelComparers.Entity.Equals(x.Entity, resource.Entity))
```

Comment:
Every resource class (aka entity) should only appear in one resource (aka aggregate), and as you alluded, only one time. I see no reason why there would be a need to check both.

Also, since you mentioned that the issue was caused by the presence of multiple null entity references (due to the implicit extension scenarios), I would suggest changing `FirstOrDefault` to `SingleOrDefault`. I think the change you're making here will fix the underlying issue, but there should only ever be at most one, so let's tighten that down so a future error is more obvious.

Reply:
Updated, thank you.